sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala

```diff
@@ -406,7 +406,7 @@ object JavaTypeInference {
               expressions.Literal(fieldName) :: extractorFor(fieldValue, fieldType) :: Nil
             })
           } else {
-            throw new UnsupportedOperationException(s"no encoder found for ${other.getName}")
+            throw new UnsupportedOperationException(s"Cannot infer type for Java class ${other.getName} because it is not bean-compliant")
```

Comment:
Thanks. I have pushed a fix.

On Thu, Jan 21, 2016 at 8:12 PM, Reynold Xin <notifications@github.com>
wrote:

> In
> sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala
> <https://github.com/apache/spark/pull/10865#discussion_r50498058>:
>
> > @@ -406,7 +406,7 @@ object JavaTypeInference {
> >                expressions.Literal(fieldName) :: extractorFor(fieldValue, fieldType) :: Nil
> >              })
> >            } else {
> > -            throw new UnsupportedOperationException(s"no encoder found for ${other.getName}")
> > +            throw new UnsupportedOperationException(s"Cannot infer type for Java class ${other.getName} because it is not bean-compliant")
>
> this needs to be less than 100 chars wide to fit spark's style guide
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/10865/files#r50498058>.
>